interface/java_binding/src/cpp/libktx-jni.cpp

```diff
@@ -0,0 +1,131 @@
+
+void set_ktx_texture(JNIEnv *env, jobject thiz, ktxTexture *texture)
+{
+    jclass ktx_texture_class = env->GetObjectClass(thiz);
+    jfieldID ktx_instance_field = env->GetFieldID(ktx_texture_class, "instance", "J");
+
+    env->SetLongField(thiz,
+                        ktx_instance_field,
+                        reinterpret_cast<jlong>(texture));
+}
+
+jobject make_ktx1_wrapper(JNIEnv *env, ktxTexture1 *texture)
+{
+    jclass ktx_texture_class = env->FindClass("org/khronos/ktx/KTXTexture1");
+    assert (ktx_texture_class != NULL);
+
+    jmethodID ktx_texture_ctor = env->GetMethodID(ktx_texture_class, "<init>", "(J)V");
+
+    return env->NewObject(ktx_texture_class, ktx_texture_ctor, reinterpret_cast<jlong>(texture));
+}
+
+jobject make_ktx2_wrapper(JNIEnv *env, ktxTexture2 *texture)
+{
+    jclass ktx_texture_class = env->FindClass("org/khronos/ktx/KTXTexture2");
+    assert (ktx_texture_class != NULL);
+
+    jmethodID ktx_texture_ctor = env->GetMethodID(ktx_texture_class, "<init>", "(J)V");
+
+    return env->NewObject(ktx_texture_class, ktx_texture_ctor, reinterpret_cast<jlong>(texture));
+}
+
+void copy_ktx_texture_create_info(JNIEnv *env, jobject info, ktxTextureCreateInfo &out)
+{
+    jclass ktx_info_class = env->GetObjectClass(info);
+    jfieldID f_gl_internalformat = env->GetFieldID(ktx_info_class, "glInternalformat", "I");
+    jfieldID f_base_width = env->GetFieldID(ktx_info_class, "baseWidth", "I");
+    jfieldID f_base_height = env->GetFieldID(ktx_info_class, "baseHeight", "I");
+    jfieldID f_base_depth = env->GetFieldID(ktx_info_class, "baseDepth", "I");
+    jfieldID f_num_dimensions = env->GetFieldID(ktx_info_class, "numDimensions", "I");
+    jfieldID f_num_levels = env->GetFieldID(ktx_info_class, "numLevels", "I");
+    jfieldID f_num_layers = env->GetFieldID(ktx_info_class, "numLayers", "I");
+    jfieldID f_num_faces = env->GetFieldID(ktx_info_class, "numFaces", "I");
+    jfieldID f_is_array = env->GetFieldID(ktx_info_class, "isArray", "Z");
+    jfieldID f_generate_mipmaps = env->GetFieldID(ktx_info_class, "generateMipmaps", "Z");
+    jfieldID f_vk_format = env->GetFieldID(ktx_info_class, "vkFormat", "I");
+
+    out.glInternalformat = env->GetIntField(info, f_gl_internalformat);
+    out.baseWidth = env->GetIntField(info, f_base_width);
+    out.baseHeight = env->GetIntField(info, f_base_height);
+    out.baseDepth = env->GetIntField(info, f_base_depth);
+    out.numDimensions = env->GetIntField(info, f_num_dimensions);
+    out.numLevels = env->GetIntField(info, f_num_levels);
+    out.numLayers = env->GetIntField(info, f_num_layers);
+    out.numFaces = env->GetIntField(info, f_num_faces);
+    out.isArray = env->GetBooleanField(info, f_is_array) ? KTX_TRUE : KTX_FALSE;
+    out.generateMipmaps = env->GetBooleanField(info, f_generate_mipmaps) ? KTX_TRUE : KTX_FALSE;
+    out.vkFormat = env->GetIntField(info, f_vk_format);
+}
+
+void copy_ktx_basis_params(JNIEnv *env, jobject params, ktxBasisParams &out)
```

Comment:
What about support for the ASTC encoder - ktxAstcParams?

Reply:
FYI @MarkCallow The `ktxTexture2_CompressAstc`, `ktxTexture2_CompressAstcEx` functions are not in the DOxygen documentation currently deployed.

Reply:
```
[ 40%] Linking CXX shared library libktx-jni.dylib
Undefined symbols for architecture x86_64:
  "_ktxTexture2_CompressAstc", referenced from:
      _Java_org_khronos_ktx_KTXTexture2_compressAstc in KTXTexture2.cpp.o
  "_ktxTexture2_CompressAstcEx", referenced from:
      _Java_org_khronos_ktx_KTXTexture2_compressAstcEx in KTXTexture2.cpp.o
```

The ASTC encoder isn't present in `libktx` for me. I don't think it's present in the latest release.

Reply:
The deployed documentation is only updated when there is a release. The ASTC encoder hasn't been released yet but it is in `master` and is built and tested in CI. This must relate to your second question and your build fails due to using the released libktx that you have installed rather than building it from source.